src/v/kafka/server/group.h

```diff
@@ -456,6 +464,18 @@ class group {
     model::term_id _term;
     absl::node_hash_map<model::topic_partition, offset_metadata>
       _pending_offset_commits;
+
+    struct volatile_offset {
+        model::offset offset;
+        int32_t leader_epoch;
+        std::optional<ss::sstring> metadata;
+    };
+
+    struct volatile_tx {
+        absl::node_hash_map<model::topic_partition, volatile_offset> offsets;
```

Comment:
not a todo for this commit, but @dotnwat , @rystsov  - i wonder if `topic_partition` and `ntp` which are really quite heavily used, if we should move them to be `value-types` say 3 integers and every core has a `model::canonical_topic_name`, but by default the `model::topic_name` is really an alias to a `int64_t` and when you print it, it does a lookup on a core-local table. 

Maybe we make a todo if this becomes a problem in perf. 

Nothing to do in this comment, but since we manipulate that data type at the heart of the system, probably worth optimizing for.

Reply:
actually, i think kafka is moving to these topic identifiers - https://cwiki.apache.org/confluence/display/KAFKA/KIP-516%3A+Topic+Identifiers

Reply:
so we should revisit then.